### ISSUE TITLE
ajax function put to data-ajax?

### ISSUE THREAD

**Issue by joomlaA1** [wenzhixin/bootstrap-table#1787]
hello @dabros
they order something like this...

function listUsers() {
var request = $.ajax({
url: "api/users/listUsers.php",
method: "POST"
})
.done(function(response) {
$('#mainContent').html(response);
})
.fail(function() {
console.log ("Could not retrieve the list of users.");
});
}

<table data-toggle="table" data-ajax="listUsers"></table>

is it possible? please guide me... i need you...

wenzhixin/bootstrap-table-examples#105

**Comment by joomlaA1** [comment]
hello @dabros thanks...

"But why custom ajax?????"
i dont know... but they would use websocket technology so that only if there is a change in database, the table would only be updated then...


i have side question....?
what event is that "while the table is in the middle of loading data"? or something like that
...would like to put a revolving animation only while the table is loading...?

**Comment by dabros** [comment]
@joomlaA1 - any url that returns json can be used, ajax is just asynchronise request and is already used. POST or GET is definable without custom data-ajax, and that url would work exact same, with `responseHandler`

The 'detecting changes' idea might mean checking via ajax before reloading, which has been discussed here: #1646.

But i wouldnt be overly concerned unless you see actual performance issues, or definable use case where forcing refresh X number of seconds would cause issues. (see linked issue)

And even then, there is a simpler workaround, as i stated similar in the linked issue. I would recommend high level variable tracking last request time and some url endpoint that returns last time data source was edited.

This local variable is then checked either using line inside request function (see topic) or via seperate ajax 'ticker' (timed interval request) that, if it detects the need, just calls `bootstrapTable('refresh')`.

This wouldn't affect data retrieval method AT ALL (so no data-ajax or other data-ajax-X options needed), just trigger refresh if needed.

The linked topic and what i posted here is all you need, just google to find any syntax mentioned but not shown in detail here or there - not hard to do but very contextual so we cant do it for you more than what you see in linked issue #1646.

---

And regarding loading icon, just look at the examples and see its already there: http://issues.wenzhixin.net.cn/bootstrap-table/

If you want to style it more, id just add style rules and or js content adjustment for '.fixed-table-loading', the element that if you hit F12 you can see is whats being toggled.

Or try looking for existing solution, like https://github.com/wenzhixin/bootstrap-table/issues?utf8=%E2%9C%93&q=is%3Aissue+load, i see a bunch there maybe of interest.

**Comment by wenzhixin** [comment]
@joomlaA1 If you want to use websocket technology, i think you can use `load`, `update`, `append`, and `remove` methods to update the table.

**Comment by joomlaA1** [comment]
hello @wenzhixin

what do you think...

they want total removal of ajaxes and just use the websocket connection (they say since its already there)

total removal of ajax also include removal of "data-url"... they say...

what you think...?


to @dabros:
side question #2 (sry... but i ask coz u r expert, can i mail you?)
in plugging any plugin to page, preferred way is by "data-x" or by javascript..? some idea is data-x is "debugging horror"? thanks @dabros

**Comment by dabros** [comment]
@joomlaA1 

Them saying its "already there" is useless to us helping you more.

We have already explained here and in other issues and in doco how to interact with `boostrapTable` for making data requests. And jquery has its own extensive doco, tutorials and help posts regarding widely varied uses of ajax.

Unless you can present the following we cannot help you:

1. Code and description of existing situation

2. Clear summary of changes you tried making but failed, and detail why to best of your knowledge

3. Clear current output vs expected output examples

4. Not least of all, show you workings!!

Show that you tried the prescribed approach and failed, or tried our suggestion and failed.

Show us why 'they' dont want to use what IS ajax code exposed very simply through `boostrapTable`, ie show us what they have already and detail how it wont work.

What i actually recommend is to find jquery guides to make this work WITHOUT bootstrap table - make a working example of 'thier' websocket code working to keep a local javascript array filled with the right data.

Ignore loading into bootstrap table totally, as this issue issnt even about the table till you get working code connecting to your server how you want - bootstrapTable does NOTHING fancy that will break any approach you choose, at most just neccessitating some callback code to play nicely with whatever you using.

As ive said in other posts, the issue here is your jquery knowledge, not boostrapTable. 

And that means this issnt the appropriate location for your posts until you know how to make things without boostrapTable - ie how to make a websocket data request that knows how to keep in sync with its remote datasource, how to manipulate html with $.append() and ect (different question you asked, same issue), what "data-x" commonly does in jquery, ect.

'They' saying to remove something we explained and both you and they show no signs of understanding is just basic jquery ajax() shows that you prob dont have a working method of doing this at all, regardless of boostrapTable, and until you can we cant help you more.

---

regarding side question no. 2, "data-x" is "debuggin horror" sounds like reference to what ive explained to you in other posts, that "data-x" is a 'toggle' that is automattically evaluated on document.ready() or similar jquery call. 

In short, you cant control the timing and that sometimes causes issues if using other dynamic type code, but in terms of straight debugging it wont 'stop' you debugging any aspect of the jquery object 'boostrapTable', but it does make it harder.

Far better to implement
